src/api/make.js

```diff
@@ -52,9 +52,29 @@ export default async (providedOptions = {}) => {
     forgeConfig = await getForgeConfig(dir);
   });
 
-  if (platform && platform !== process.platform && !(process.platform === 'darwin' && platform === 'mas')) {
-    throw "You cannot run 'make' for a platform other than your system's platform";
-  }
+  const makers = {};
+  const targets = overrideTargets || forgeConfig.make_targets[platform];
+
+  targets.forEach((target) => {
+    const maker = requireSearchRaw(__dirname, [
+      `../makers/${platform}/${target}.js`,
+      `../makers/generic/${target}.js`,
+      `electron-forge-maker-${target}`,
+      target,
+      path.resolve(dir, target),
+      path.resolve(dir, 'node_modules', target),
+    ]);
+
+    if (!maker) {
+      throw `Could not find a build target with the name: ${target} for the platform: ${platform}`;
+    }
+
+    if (!maker.supportsPlatform) {
+      throw `Cannot build for ${platform} target ${target} from your ${process.platform} device`;
```

Comment:
You make a good point 👍 

Perhaps this would work
* If there is a function call it and check
* If there is not a function throw an error saying the maker is incompatible with the current version of forge and to ask the creator to update their maker